### PR TITLE
[NHP-7] Implement Vite bundler

### PR DESCRIPTION
**Pull request by exxocism**
## JIRA issue/Description
- JIRA: https://exxocism.atlassian.net/browse/NHP-7
- Implements `vite` bundler to make a runtime

## Major changes
- Vite is added : [657c0f8](https://github.com/exxocism/node-healthz-prober/pull/3/commits/657c0f87efabe8300653f0fef9e663534522926a), [6658688](https://github.com/exxocism/node-healthz-prober/pull/3/commits/66586884f7badacaf00ecd19c13ce4f6b14f8873), [09e5da0](https://github.com/exxocism/node-healthz-prober/pull/3/commits/09e5da041921aa9fc1964b1f973dca4e140b2211)
- Module type is changed from `esm` to `commonJS`: [b5120df](https://github.com/exxocism/node-healthz-prober/pull/3/commits/b5120df2206bfc675a8db83e02c4bee12434f833)

### 1. local development
- Install `node.js` by nvm: [windows](https://github.com/coreybutler/nvm-windows), [linux/mac](https://github.com/nvm-sh/nvm)
- Use node `20.10.0`: `nvm use`
- Install the required files: `npm ci`
### 4. How to test
- [ ] Check if `npm start` works properly
  ```
  Hello, world!
  ```
- [ ] Create a `.env` file with `VITE_` prefix 
  ```env
  # inside .env file
  VITE_TEST_ENV=test
  ```
- [ ] Change `index.ts` to use `VITE_` prefix environment variable
  https://github.com/exxocism/node-healthz-prober/blob/09e5da041921aa9fc1964b1f973dca4e140b2211/src/index.ts#L1
  ```tsx
  console.log('Hello, world!', process.env.VITE_TEST_ENV);
  ```
- [ ] Run `npm run build`
- [ ] Visit `dist` folder and if the `VITE_TEST_ENV` is embedded into `index.js`
  ```tsx
  var e={VITE_TEST_ENV:"test"};console.log("Hello, world!",e.VITE_TEST_ENV);
  ```
- [ ] Run `npm start` and check if it shows `VITE_TEST_ENV` environment variable
  ```
  Hello, world! test
  ```
- [ ] Run `npm run runtime` and check if it builds the executable successfully
  - Linux/Mac `dist/executable`, Windows: `dist/executable.exe`
- [ ] Check if all the commands work in the different platform
  - [ ] Windows
  - [ ] Linux
  - [ ] MacOS

## TODO
- Pre-deployment steps: None
- Post-deployment steps:
  - [ ] Known issue: `npm run dev` is not wokring now: https://exxocism.atlassian.net/browse/NHP-8

### CODE REVIEW

File: vite.config.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+          manualChunks: false,
+          inlineDynamicImports: true,
+          entryFileNames: '[name].js',
+          assetFileNames: '[name].[ext]',
+        },
+      },
+    },
+    define: {
+      'process.env': { ...loadEnv(mode, process.cwd()) },
+    },
```
[the remark this answers is not in the document]
Comment (lasagnaslave) [reply vite.config.js:21]: Let's investigate it as an engineering excellence!

**Review by lasagnaslave — DISMISSED** [review]
Thanks for the effort!
Tested and confirmed working. LGTM with minor comments / questions 👍

File: tsconfig.json [hunk 2]
```diff
@@ -1,9 +1,10 @@
 {
   "compilerOptions": {
     "target": "es2017",
-    "module": "Node16",
+    "module": "CommonJS",
```
[the remark this answers is not in the document]
Comment (exxocism) [reply tsconfig.json:4]: Great suggestion! I created a jira card for this: https://exxocism.atlassian.net/browse/NHP-12

File: package.json [hunk 3]
```diff
@@ -3,10 +3,11 @@
   "version": "1.0.0",
   "description": "probes healthz endpoints and reports to slack",
   "main": "src/index.ts",
+  "type": "module",
```
[the remark this answers is not in the document]
Comment (exxocism) [reply package.json:6]: https://github.com/exxocism/node-healthz-prober/pull/3#discussion_r1420309947

File: vite.config.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { defineConfig, loadEnv } from 'vite';
+import commonjs from '@rollup/plugin-commonjs';
+
+export default ({ mode }) => {
+  return defineConfig({
+    build: {
+      rollupOptions: {
+        plugins: [commonjs()],
+        // overwrite default .html entry
```
[the remark this answers is not in the document]
Comment (exxocism) [reply vite.config.js]: Thanks for the catch! Applied in [f5d6b6e](https://github.com/exxocism/node-healthz-prober/pull/3/commits/f5d6b6eb355a36a7c9f44eb32661a2605e7df8b3)

**Review by lasagnaslave — APPROVED** [review]

**Pull request merged**